upload-api/buckets/car-store.js

```diff
@@ -103,3 +103,55 @@ export function useCarStore(s3, bucketName) {
     },
   }
 }
+
+/**
+ * compose many car stores.
+ * store#createUploadUrl is from first store.
+ * store#has will check stores in order until 0-1 `true` are found.
+ * 
+ * @param  {import('@web3-storage/upload-api').CarStoreBucket} carStore 
+ * @param  {Array<import('@web3-storage/upload-api').CarStoreBucket>} moreCarStores 
+ */
+export function composeCarStoresWithOrderedHas(carStore, ...moreCarStores) {
```

Comment:
if this name still feels wrong after @vasco-santos pointed it out, honestly happy to just accept the first suggestion